src/Microsoft.Health.Dicom.Blob/Features/Storage/BlobFileStore.cs

```diff
@@ -123,13 +136,50 @@ public class BlobFileStore : IFileStore
         return fileProperties;
     }
 
+    // TODO: This should removed once we migrate everything and the global flag is turned on
     private BlockBlobClient GetInstanceBlockBlob(VersionedInstanceIdentifier versionedInstanceIdentifier)
     {
-        string blobName = $"{versionedInstanceIdentifier.StudyInstanceUid}/{versionedInstanceIdentifier.SeriesInstanceUid}/{versionedInstanceIdentifier.SopInstanceUid}_{versionedInstanceIdentifier.Version}.dcm";
+        string blobName;
+        if (_supportNewBlobFormatForNewService)
+        {
+            blobName = _nameWithPrefix.GetInstanceFileName(versionedInstanceIdentifier);
+        }
+        else
+        {
+            blobName = _nameWithUid.GetInstanceFileName(versionedInstanceIdentifier);
+        }
 
         return _container.GetBlockBlobClient(blobName);
     }
 
+    private BlockBlobClient[] GetInstanceBlockBlobs(VersionedInstanceIdentifier versionedInstanceIdentifier)
```

Comment:
This naming seems weird. Seems like this method (and the existing one) should be `GetInstanceBlockBlobClients`

Reply:
Updated